### DISCUSSION TITLE
Will there be a Native AOT in .NET

### DISCUSSION THREAD

**Discussion by vishvanathan-k** [dotnet/runtime#48266] [General] [answered]
Hi , 
Will .NET 6 or .NET 5.x will come with an Full , Native AOT for all projects i.e. from console to WPF to ASP.NET to Blazor
PS: I know about .NET Native

**Comment by ryancheung** [comment, marked as answer]
See https://github.com/dotnet/runtimelab/tree/feature/NativeAOT

**Reply by vishvanathan-k** [reply]
Thanks!

**Reply by vasicvuk** [reply]
The question is will AOT be supported in .NET 6 as LTS feature or only partial experimental support?

**Reply by svick** [reply]
@vasicvuk [The .Net 6 roadmap](https://themesof.net/?q=is:open%20kinds:teu%20release:%22.NET%206.0%22) includes *Early adopters can experiment with native AOT form factor* https://github.com/dotnet/runtimelab/issues/248 (emphasis on "experiment"), so I think there is no chance it will be supported when .Net 6 is released later this year.

**Reply by davidfowl** [reply]
No, not in .NET 6

**Reply by vishvanathan-k** [reply]
What !!! So NO NATIVE AOT or any other AOT Compilation for .NET 6 execpt blazor . :disappointed:  Can't they atleast update .net native to support .net 6 , that will atleast easier

**Reply by juepiezhongren** [reply]
mono aot is ok with blazor. u should know mono llvm is a proven and tested runtime, where native aot is just starting. eventually we want core and mono completely merge into one.
